socorro/unittest/external/es/test_super_search_fields.py

```diff
@@ -40,93 +40,95 @@ def test_get_fields(self):
+        }
+
+        fake_mapping_2 = {
+            config.elasticsearch_doctype: {
+                "properties": {
+                    "namespace1": {
+                        "type": "object",
+                        "properties": {
+                            "subspace1": {
+                                "type": "object",
+                                "properties": {"field_d": {"type": "long"}},
+                            }
+                        },
                     }
                 }
-            },
-        ]
+            }
+        }
+
+        # Refresh and then delete existing indices so we can rebuild the mappings
+        # in order to diff them
+        self.es_context.refresh()
+        for index_name in self.es_context.get_indices():
+            self.es_context.delete_index(index_name)
+
+        # Refresh ES to wait for indices to delete
+        self.es_context.refresh()
 
         now = datetimeutil.utc_now()
-        indices = []
-
-        try:
-            # Using "2" here means that an index will be missing, hence testing
-            # that it swallows the subsequent error.
-            for i in range(2):
-                date = now - datetime.timedelta(weeks=i)
-                index = date.strftime(api.context.get_index_template())
-                mapping = fake_mappings[i % len(fake_mappings)]
-
-                api.context.create_index(index, mappings=mapping)
-                indices.append(index)
-
-            api = SuperSearchFieldsModel(config=config)
-            missing_fields = api.get_missing_fields()
-            expected = [
-                "field_z",
-                "namespace1.field_a",
-                "namespace1.field_b",
-                "namespace1.subspace1.field_d",
-                "namespace2.subspace1.field_b",
-            ]
-
-            assert missing_fields["hits"] == expected
-            assert missing_fields["total"] == 5
-
-        finally:
-            for index in indices:
-                self.index_client.delete(index=index)
+        template = api.context.get_index_template()
+
+        # Create an index for now
+        index_name = now.strftime(template)
+        mapping = fake_mapping_1
+        api.context.create_index(index_name, mappings=mapping)
+
+        # Create an index for 7 days ago
+        index_name = (now - timedelta(days=7)).strftime(template)
+        mapping = fake_mapping_2
+        api.context.create_index(index_name, mappings=mapping)
+
+        # Refresh ES to wait for indices to be created
+        self.es_context.refresh()
+
+        api = SuperSearchFieldsModel(config=config)
+        missing_fields = api.get_missing_fields()
+        expected = [
+            "field_z",
+            "namespace1.field_a",
+            "namespace1.field_b",
+            "namespace1.subspace1.field_d",
+            "namespace2.subspace1.field_b",
+        ]
+
+        assert missing_fields["hits"] == expected
+        assert missing_fields["total"] == 5
```

Comment:
We're doing deletion in teardown now. The rest of these changes are mostly dedenting and making the test clearer without iterating over a range of 0, 1.